### PR TITLE
Introduce the CalendarEventAttendance/get JMAP method

### LINKED ISSUE #1328: Provide a JMAP method: CalendarEventAttendance/get

### PR DESCRIPTION
**Pull request by HoussemNasri**
Closes #1328 

## TODO
- [x] Integration tests
- [x] Documentation

### CODE REVIEW

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala [hunk 1]
```diff
@@ -0,0 +1,308 @@
...
+import org.apache.http.HttpStatus.SC_OK
+import org.apache.james.GuiceJamesServer
+import org.apache.james.jmap.http.UserCredential
+import org.apache.james.jmap.rfc8621.contract.Fixture._
+import org.apache.james.mailbox.model.MailboxPath
+import org.apache.james.modules.MailboxProbeImpl
+import org.apache.james.utils.DataProbeImpl
+import org.junit.jupiter.api.{BeforeEach, Test}
+
+trait LinagoraCalendarEventAttendanceGetMethodContract {
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala:18]: Please test edge cases:
 - invalid blob
 - blob is not an ICS
 - missing capability
 - etc...

File: tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepositoryTest.scala [hunk 2]
```diff
@@ -199,4 +205,12 @@ class StandaloneEventAttendanceRepositoryTest {
...
     new MessageIdManagerTestSystem(resources.getMessageIdManager, messageIdFactory, resources.getMailboxManager.getMapperFactory, resources.getMailboxManager)
   }
+
+  private def done(blobId: BlobId, attendanceStatus: AttendanceStatus) =
+    CalendarEventAttendanceResults.done(
+      EventAttendanceStatusEntry(
+        blobId.value.value,
+        attendanceStatus
+      )
+    )
```
Comment (chibenwa) [inline tmail-backend/jmap/extensions/src/test/scala/com/linagora/tmail/james/jmap/StandaloneEventAttendanceRepositoryTest.scala:215]:
```suggestion
        attendanceStatus))
```

**Review by chibenwa — COMMENTED** [review]
Good quality work.

Nothing much to say appart that things starts to piece nicely together ;-)

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala [hunk 3]
```diff
@@ -228,8 +229,272 @@ trait LinagoraCalendarEventAttendanceGetMethodContract {
...
+         |    "urn:ietf:params:jmap:core",
+         |    "com:linagora:params:calendar:event"],
+         |  "methodCalls": [[
+         |    "CalendarEventAttendance/get",
+         |    {
+         |      "accountId": "$ANDRE_ACCOUNT_ID",
+         |      "blobIds": [ "$blobId" ]
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala:338]: Tell me more about this please.

What is the actual behaviour?
Can we be sure Bob can do `CalendarEventAttendance/get` calls if, and only of the mailbox containing the blobs had been delegated to him? (Otherwise it is a -minor- security vulnerability)

Comment (HoussemNasri) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala:338]: The problem boils down to `StandaloneEventAttendanceRepository#getFlags`.

When logging the returned message, it is actually returned even when the session user does not own that message.

Comment (HoussemNasri) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala:338]: It is a false alarm, I probably misread the output of one test case to the permission one.

File: tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala [hunk 4]
```diff
@@ -317,7 +318,6 @@ trait LinagoraCalendarEventAttendanceGetMethodContract {
            |]""".stripMargin)
   }
 
-  @Disabled("A system session is used to fetch message flags from user inbox which does not seem to handle user mailbox isolation.")
   @Test
```
Comment (chibenwa) [inline tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala]: Please add one test

```
GIVEN Alice delegated her INBOX to BOB
AND Alice has an ICS in her INBOX
WHEN bob do EventAttendance/get
THEN XXXX
```

(We need to document the current behaviour...

Comment (HoussemNasri) [reply tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraCalendarEventAttendanceGetMethodContract.scala]: It succeeds https://github.com/linagora/tmail-backend/pull/1447/commits/8e97b5a7842e4c85f2e488af560b9c068a864f55

**Comment by chibenwa** [comment]
Now, the dav version with module chooser :-)

**Comment by Arsnael** [comment]
Please clean up commit history and add the issue as a prefix on commit descriptions
